metals/src/main/scala/scala/meta/internal/metals/OnTypeFormattingProvider.scala

```
@@ -0,0 +1,71 @@
+import scala.meta.tokens.Token
+import scala.meta.tokens.Token.Constant
+
+final class OnTypeFormattingProvider(
+    semanticdbs: Semanticdbs
+)(implicit ec: ExecutionContext) {
+
+  val quot = """\u0022\u0022\u0022"""
+  val space = " "
+
+  private def indent(toInput: Input.VirtualFile, pos: meta.Position): String = {
+    val bit = toInput.text.substring(0, pos.start)
+    val lastNewline = bit.lastIndexOf("\n")
+    val lastPipe = bit.lastIndexOf("|")
+    val length = toInput.text
+      .substring(lastNewline, lastPipe)
+      .length - 1
+    space * length
+  }
+
+  private def longString(text: String, token: Token): Boolean = {
+    text.substring(token.start, token.start + 3).equals(quot)
+  }
+
+  private def inToken(pos: meta.Position, token: Token): Boolean = {
+    pos.start >= token.start && pos.end <= token.end
+  }
+
+  private def pipeInScope(pos: meta.Position, text: String): Boolean = {
+    val lastNew =
+      text.substring(0, pos.start).lastIndexOf("\n")
+    text
+      .substring(lastNew, pos.start)
+      .contains("|")
+  }
+
+  def format(
+      params: DocumentOnTypeFormattingParams
+  ): Future[Option[TextEdit]] = {
+    val source = params.getTextDocument.getUri.toAbsolutePath
+    if (source.exists) {
+      val range = new Range(params.getPosition, params.getPosition)
+      val pos = params.getPosition.toMeta(source.toInput)
+      val sourceText = source.toInput.text
+      val tokens = source.toInput.tokenize.toOption
+      var edit: Option[TextEdit] = None
+      tokens.get.foreach {
+        case token @ Constant.String(_) =>
+          if (inToken(pos, token) && longString(sourceText, token) && pipeInScope(
+              pos,
+              sourceText
+            ))
+            edit = Some(new TextEdit(range, indent(source.toInput, pos) + "|"))
+        case _ => None
+      }
+      Future.successful(edit)
+    } else {
+      Future.failed(new Exception("There is no source"))
```

Why failure instead of a `None` ?